src/main/java/seedu/address/model/attendance/AttendanceList.java

```diff
@@ -70,6 +70,12 @@ public void generateAttendance(UniqueStudentsInTutorialList uniqueStudentList) {
                                 new Comment("")));
             }
         }
+
+        for (Attendance attendance : attendances) {
+            if (!uniqueStudentList.containsStudentWithId(attendance.getStudentId())) {
+                attendances.remove(attendance);
+            }
+        }
```

Comment:
remove student command calls generate attendance, which will remove all non-students of the tutorial from the attendance list.